src/utilities/modules/temperatureSensorHelpers.js

```diff
@@ -370,87 +382,170 @@ function addHeadAndTrailingLogToBreach({ result, sensor, database }) {
+    .max('timestamp');
+  // Starting timestamp is either from the first fullAggregate timestamp, or beginning of time
+  const aggregationIntervalStart = (mostRecentFullAggregateTimestamp || -Infinity) + 1;
+  // Ending timestamp is either from the most recent preAggregate or from now, less the
+  // no full aggregate period (32 hours)
+  const aggregationIntervalEnd =
+    (mostRecentPreAggregateTimestamp || new Date().getTime()) - NO_FULL_AGGREGATE_PERIOD;
+  // Query for all sensorLogs which are preAggregates, between the above two
+  // timestamps
+  const sortedSensorLogs = sensorLogs
+    .filtered(
+      'aggregation == $0  && timestamp > $1 && timestamp < $2',
+      SENSOR_LOG_PRE_AGGREGATE_TYPE,
+      aggregationIntervalStart,
+      aggregationIntervalEnd
+    )
+    .sorted('timestamp');
+  // Will hold the timestamp of a sequential group of preAggregate logs
+  // which are to-be aggregated into a fullAggregate.
+  let logGroupStartTimestamp = Infinity;
+  // sensorLogGroups is a 2D array for all groups of to-be-aggregated sensor logs.
+  // sensorLogGroup is A collection of sensor logs which will be aggregated into a
+  // full aggregated sensorLog - 1D arrays housed within the 2D array.
+  const sensorLogGroups = [];
+  let sensorLogGroup = [];
+  // Find sets of sensorLogs within the full aggregation interval. Delimited by
+  // a sensorLog which is not within the same interval.
+  sortedSensorLogs.forEach(sensorLog => {
+    const { timestamp } = sensorLog;
+    // Premature return if this sensorlog is malformed
+    if (!timestamp) return;
+    // Check if this sensorLog is in the current interval. The first iteration
+    // will never be in the same interval, as one hasn't been created/started.
+    const isInSameInterval = logGroupStartTimestamp - timestamp < FULL_AGGREGATION_INTERVAL;
+    // If so, add it to the current group of logs.
+    if (isInSameInterval) sensorLogGroup.push(sensorLog);
+    // Otherwise, hit a delimiter. If there are already logs in the current
+    // sensor log group, this is an ending delimiter, store the group and reset.
+    else if (sensorLogGroup.length !== 0) sensorLogGroups.push([...sensorLogGroup]);
+    // Resetting for a new group of sensor log. This is called on the first iteration
+    // to initialize the first group. The last group of sensorLogs will not be set aside
+    // for aggregation unless it is also delimited by a sensorLog not within that group.
+    if (!isInSameInterval) {
+      sensorLogGroup = [sensorLog];
+      logGroupStartTimestamp = timestamp;
+    }
   });
-
+  // Create an array of aggregated sensor log objects
+  const aggregatedSensorLogs = sensorLogGroups
+    .map(group => createFullAggregateSensorLogs(group))
+    .flat();
+  const deleteFromTimestamp =
+    sensorLogGroup.length > 0 ? sensorLogGroup[0].timestamp : aggregationIntervalEnd;
+  const sensorLogsToDelete = sortedSensorLogs.filtered('timestamp < $0', deleteFromTimestamp);
```

Comment:
@joshxg, i think this is not quite right yet, shouldn't it it just be sensorLogGroup.flat() ?

Reply:
`sensorLogGroup` is a 1D array, filled with sensorLog objects, not sure what you mean. Essentially it is a stack which builds an 8 hour interval of `sensorLogs`. If we take the first log from that, it is the earliest in the last set of `sensorLogs` which have not been aggregated yet and we can delete all the pre aggregate logs left over and delete them, as we have already filtered `sortedSensorLogs` to be a collection of `sensorLogs` that are to-be-aggregated. If `sensorLogGroup` does not have any sensorLogs in it, we can just delete everything as there are no more `sensorLogs` to aggregate (this case shouldn't occur, maybe it is to safe?)

Reply:
@joshxg I missed the 's' on sensorLogGroups.
`sensorLogsToDelete = sensorLogGroups.flat()`
 
Your code all good apart from when 0 sortedSensorLogs?

Reply:
Oh right, sorry. I think I was not 100% that actual realm objects were required or not for the delete method.

Technically the method does nothing when `sortedSensorLogs.length = 0`, but can add an early return.

On that note, I forgot to ask if you think the early return on no `sensorLogs` should return an object in the same form, or if null is fine. Further, I kinda hate the result parameter and return value, heh

Reply:
@joshxg I think you need to loop through each element to delete it of not realm array. I thought that would be cleaner and quicker then query realm in this case

Reply:
I am not a fan of returns either, can discuss tomorrow